qa/common/gen_qa_plugin_models.py

```diff
@@ -0,0 +1,180 @@
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import argparse
+from builtins import range
+import os
+import sys
+
+import numpy as np
+import tensorrt.legacy as trt
+
+TRT_LOGGER = trt.Logger()
+
+trt.init_libnvinfer_plugins(TRT_LOGGER, '')
+PLUGIN_CREATORS = trt.get_plugin_registry().plugin_creator_list
+
+def np_to_model_dtype(np_dtype):
+    if np_dtype == np.bool:
+        return "TYPE_BOOL"
+    elif np_dtype == np.int8:
+        return "TYPE_INT8"
+    elif np_dtype == np.int16:
+        return "TYPE_INT16"
+    elif np_dtype == np.int32:
+        return "TYPE_INT32"
+    elif np_dtype == np.int64:
+        return "TYPE_INT64"
+    elif np_dtype == np.uint8:
+        return "TYPE_UINT8"
+    elif np_dtype == np.uint16:
+        return "TYPE_UINT16"
+    elif np_dtype == np.float16:
+        return "TYPE_FP16"
+    elif np_dtype == np.float32:
+        return "TYPE_FP32"
+    elif np_dtype == np.float64:
+        return "TYPE_FP64"
+    elif np_dtype == np_dtype_string:
+        return "TYPE_STRING"
+    return None
+
+def np_to_trt_dtype(np_dtype):
+    if np_dtype == np.int8:
+        return trt.infer.DataType.INT8
+    elif np_dtype == np.int32:
+        return trt.infer.DataType.INT32
+    elif np_dtype == np.float16:
+        return trt.infer.DataType.HALF
+    elif np_dtype == np.float32:
+        return trt.infer.DataType.FLOAT
+    return None
+
+def get_trt_plugin(plugin_name):
+        plugin = None
+        for plugin_creator in PLUGIN_CREATORS:
+            if plugin_creator.name == plugin_name:
```

Comment:
The following seems to be "LReLU_TRT" specific? What if you pass other plugin's name which is also in plugin_creator?

Reply:
We can create models that use other plugins but since all plugins are registered in one step, it is safe to say that if one plugin works then the others will as well.

Reply:
I understand that. What I was saying is that `get_trt_plugin` is rather general name and it accepts `plugin_name` which imply that it will returns the plugin that corresponds to the name, while the actual implementation is specific to ReLu...